ddtrace/contrib/flask/patch.py

```diff
@@ -560,15 +576,22 @@ def _traced_request(pin, wrapped, instance, args, kwargs):
         # This call may be unnecessary since we try to add the tags earlier
         # We just haven't been able to confirm this yet
         _set_request_tags(span)
-
-        with pin.tracer.trace(
-            ".".join(("flask", name)), service=trace_utils.int_service(pin, config.flask, pin)
-        ) as request_span:
-            # set component tag equal to name of integration
-            request_span.set_tag_str("component", config.flask.integration_name)
-
-            request_span._ignore_exception(werkzeug.exceptions.NotFound)
-            return wrapped(*args, **kwargs)
+        request = flask.request
+        _DD_EARLY_IP_CONTEXTVAR.set(request.remote_addr)
+        _DD_EARLY_HEADERS_CONTEXTVAR.set(request.headers)
+
+        try:
+            with pin.tracer.trace(
+                ".".join(("flask", name)),
+                service=trace_utils.int_service(pin, config.flask, pin),
+            ) as request_span:
+                # set component tag equal to name of integration
+                request_span.set_tag_str("component", config.flask.integration_name)
+
+                request_span._ignore_exception(werkzeug.exceptions.NotFound)
+                return wrapped(*args, **kwargs)
+        finally:
+            _reset_contextvars()
```

Comment:
Yep. It's in the appsec package so it's namespaced but since we usually do symbolic imports it'll be better to rename it to get a clearer picture that is an ASM thing when reading the code. Didn't want to complicate it much since it's supossed to be a temporary thing but I'll take your suggestion.

Commited.